### PR TITLE
chore(android): optimize proxy constructor lookup

### PR DESCRIPTION
**Pull request by garymathews**
- Minor optimization for V8 proxy constructor lookup
- Prevent unnecessary `GetPropertyNames()` call
- Correctly sets prototype constructor
  - This means `exports->GetConstructorName()` returns the proxy name

##### TEST CASE
```JS
const win = Ti.UI.createWindow();
const label = Ti.UI.createLabel();
const label_derived = new label.constructor({ text: 'This is derived from a constructor.' });

console.log(`label.constructor: ${label.constructor}`);
console.log(`label_derived.constructor: ${label_derived.constructor}`);

win.add(label_derived);
win.open();
``` 

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-28437)

### CODE REVIEW

File: android/kroll-apt/src/main/resources/org/appcelerator/kroll/annotations/generator/ProxyBindingV8.cpp.fm [hunk 1]
```diff
@@ -70,6 +70,7 @@ void ${className}::bindProxy(Local<Object> exports, Local<Context> context)
 	moduleInstance.Reset(isolate, instance);
 	<#else>
 	exports->Set(context, nameSymbol, constructor);
```
Comment (garymathews) [inline android/kroll-apt/src/main/resources/org/appcelerator/kroll/annotations/generator/ProxyBindingV8.cpp.fm:72]: NOTE: Can't remove this line as it's necessary for modules

**Comment by jquick-axway** [comment]
@garymathews, this change somehow prevents our "bootstrap.json" from being read on app startup. It happens to both encrypted and unencrypted builds. The following error will be logged, which you can see in Jenkins' output. I'm seeing this error on my end too.
`Failed to read "bootstrap.json". Reason: Unexpected token u in JSON at position 0`

**Comment by jquick-axway** [comment]
This causes our "hyperloop-examples" sample app to crash.
```
[ERROR] TiExceptionHandler: (main) [37,513] /hyperloop/android.app.Activity.js:28
[ERROR] TiExceptionHandler: 		result = Activity.class.newInstance(arguments);
[ERROR] TiExceptionHandler:                           ^
[ERROR] TiExceptionHandler: TypeError: Activity.class.newInstance is not a function
[ERROR] TiExceptionHandler:     at new Activity (/hyperloop/android.app.Activity.js:28:27)
[ERROR] TiExceptionHandler:     at Object.isInstalledViaGooglePlay (/google-play-utils.js:10:16)
[ERROR] TiExceptionHandler:     at Window.onOpen (/alloy/controllers/index.js:122:59)
[ERROR] TiExceptionHandler:     at Window.value (ti:/kroll.js:1604:27)
[ERROR] TiExceptionHandler:     at Window.value (ti:/kroll.js:1656:25)
[ERROR] TiExceptionHandler:
[ERROR] TiExceptionHandler:     org.appcelerator.kroll.runtime.v8.V8Object.nativeFireEvent(Native Method)
[ERROR] TiExceptionHandler:     org.appcelerator.kroll.runtime.v8.V8Object.fireEvent(V8Object.java:63)
[ERROR] TiExceptionHandler:     org.appcelerator.kroll.KrollProxy.doFireEvent(KrollProxy.java:985)
[ERROR] TiExceptionHandler:     org.appcelerator.kroll.KrollProxy.handleMessage(KrollProxy.java:1219)
[ERROR] TiExceptionHandler:     org.appcelerator.titanium.proxy.TiViewProxy.handleMessage(TiViewProxy.java:260)
[ERROR] TiExceptionHandler:     ti.modules.titanium.ui.WindowProxy.handleMessage(WindowProxy.java:481)
[ERROR] TiExceptionHandler:     android.os.Handler.dispatchMessage(Handler.java:102)
[ERROR] TiExceptionHandler:     android.os.Looper.loop(Looper.java:223)
[ERROR] TiExceptionHandler:     android.app.ActivityThread.main(ActivityThread.java:7664)
[ERROR] TiExceptionHandler:     java.lang.reflect.Method.invoke(Native Method)
[ERROR] V8Exception: Exception occurred at /hyperloop/android.app.Activity.js:28: Uncaught TypeError: Activity.class.newInstance is not a function
```

**Review by jquick-axway — APPROVED** [review]
CR/FR: Pass

Tested the following:
- kitchensink-v2
- hyperloop-examples
- Building module "ti.map" with this PR and using it in a 10.1.0 and 9.1.0 built app.

**Review by ewanharris — APPROVED** [review]
FR Pass, in addition to Josh's testing I also tested webpack and liveview usage

**Pull request merged**